### PR TITLE
Enable way of manually capturing errors in sentry

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Currently we wrap the `hydrate` function with `Raven.context`,  this essentially instructs Raven to wrap `hydrate` in a `try...catch` block to record any exceptions before re-throwing them. However if an error is thrown in asynchronous code such as a `Promise` chain then the error is thrown outside of this main context and not captured by Raven.

I've therefore introduced a new module `reportError.js` that exports a function `reportError`. We can now manually instruct Raven to capture exceptions by calling `reportError`, You can see examples where I've added this call to the `catch` blocks of `Promise` chains in `MostViewed.tsx` and `ShareCount.tsx`. This is simply a copy of the `report-error.js` module in `frontend`.

Explanation of the `shouldThrow` argument accepted by `reportError` can be found here https://github.com/guardian/frontend/pull/10002

Other changes include:
- I've renamed `frontend/web/lib` to `frontend/web/browser`
- I've also removed the `appData.ts` and `appData.test.ts`, used for accessing `window.guardian.app.data`. A side affect of simply importing this was it attempted to reference the `window` which caused a runtime error on the server. I've decided to remove it for now until we take a closer look at the `window.guardian` data model and how best to access it in clientside scripts.

### CODE REVIEW

File: packages/frontend/web/client/reportError.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    shouldThrow: boolean = true,
+): void => {
+    getRaven().then(raven => {
+        raven.captureException(err, { tags });
+
+        if (shouldThrow) {
+            // Flag to ensure it is not reported to Sentry again via global handlers
+            const error = err;
+            error.reported = true;
+            throw error;
```
Comment (SiAdcock) [inline packages/frontend/web/client/reportError.ts]: What are we doing to ensure Sentry doesn't report the error again via global handlers? In frontend, [we override the global `onerror` handler](https://github.com/guardian/frontend/blob/ed8d4e202b1da36d6e0454d831ff506082144b26/static/src/javascripts/bootstraps/standard/main.js#L152-L157)

**Comment by SiAdcock** [comment]
What do you think about renaming the `client` directory to `browser`? It's more explicit, in my opinion, that the code contained within will only be executed in the browser. `client` is more ambiguous

**Comment by AWare** [comment]
I think there's a place for both `lib` and `client`/`browser`

**Comment by GHaberis** [comment]
@SiAdcock @AWare all comments have been addressed

File: packages/frontend/web/browser.ts [hunk 2]
```diff
@@ -41,27 +40,65 @@ const go = () => {
...
-                    {
-                        tags: {
-                            feature: 'dotcom-rendering',
-                        },
-                    },
-                    hydrate,
-                );
+    getRaven()
+        .catch(err => {
+            hydrate();
```
Comment (SiAdcock) [inline packages/frontend/web/browser.ts]: Just thought... if we `catch` here, then `hydrate()`, we'll drop into the `then` callback and `hydrate()` again. Perhaps we should just `console.log(err)` or do nothing at this point?

**Pull request merged**